src/components/MapView/DateSelector/index.tsx

```diff
@@ -90,48 +208,105 @@ function DateSelector({ availableDates = [], classes }: DateSelectorProps) {
+          <Grid className={classes.dateContainer} ref={timeLine}>
+            <Draggable
+              axis="x"
+              handle="#timeline"
+              bounds={{
+                top: 0,
+                bottom: 0,
+                right: 0,
+                left: timeLineWidth - dateRange.length * 10,
+              }}
+              position={timelinePosition}
+              onStop={onTimelineStop}
+            >
+              <div className={classes.timeline} id="timeline">
+                <Grid
+                  container
+                  alignItems="stretch"
+                  className={classes.dateLabelContainer}
+                >
+                  {dateRange.map((date, index) => (
+                    <Tooltip
+                      title={date.label}
+                      key={date.label}
+                      TransitionComponent={Fade}
+                      TransitionProps={{ timeout: 0 }}
+                      placement="top"
+                      arrow
+                    >
+                      <Grid
+                        item
+                        xs
+                        className={
+                          date.isFirstday
+                            ? classes.dateItemFull
+                            : classes.dateItem
+                        }
+                      >
+                        {date.isFirstday ? (
+                          <Typography className={classes.dateItemLabel}>
+                            {date.month}
+                          </Typography>
+                        ) : (
+                          <div className={classes.dayItem} />
+                        )}
+                        {availableDates
+                          .map(availableDate =>
+                            moment(availableDate + USER_DATE_OFFSET).format(
+                              'DD MMM YYYY',
+                            ),
+                          )
+                          .includes(date.label) && (
+                          <div
+                            className={classes.dateAvailable}
+                            role="presentation"
+                            onClick={() => clickDate(index)}
+                          />
+                        )}
+                      </Grid>
+                    </Tooltip>
+                  ))}
                 </Grid>
-              </Fragment>
-            ))}
+                <Draggable
+                  axis="x"
+                  handle="#datePointerSelector"
+                  bounds={{
+                    top: 0,
+                    bottom: 0,
+                    left: 0,
+                    right: dateRange.length * 10,
+                  }}
+                  grid={[10, 1]}
+                  position={pointerPosition}
+                  onStart={(e: any) => e.stopPropagation()}
+                  onStop={onPointerStop}
+                >
+                  <div className={classes.triangle} id="datePointerSelector" />
```

Comment:
the datepoint selector ID should be a constant string since its used in more than one place. Also, making this a MUI icon may help with making things look more like its from MUI.